packages/db/fauna/resources/Function/findUploads.js

```diff
@@ -0,0 +1,62 @@
+import fauna from 'faunadb'
+
+const {
+  Function,
+  CreateFunction,
+  Query,
+  Lambda,
+  Match,
+  Index,
+  Var,
+  Exists,
+  If,
+  Update,
+  Paginate,
+  Ref,
+  Collection,
+  Select,
+  Filter,
+  LT,
+  Map,
+  Get,
+  Equals,
+  Let
+} = fauna
+
+const name = 'findUploads'
+const body = Query(
+  Lambda(
+    ['where', 'size', 'after', 'before'],
+    Let(
+      {
+        match: Filter(
+          Match(
+            Index('upload_sort_by_created_desc'),
+            Ref(Collection('User'), Select('user', Var('where'))),
+            true
+          ),
+          Lambda(
+            ['created', 'ref'],
+            LT(Var('created'), Select('before', Var('where')))
+          )
+        ),
+        page: If(
+          Equals(Var('before'), null),
+          If(
+            Equals(Var('after'), null),
+            Paginate(Var('match'), { size: Var('size') }),
+            Paginate(Var('match'), { size: Var('size'), after: Var('after') })
+          ),
+          Paginate(Var('match'), { size: Var('size'), before: Var('before') })
```

Comment:
Do we only support before or after?

Reply:
Yes, this is slightly confusing.

The top level `before` and `after` are cursor values that can also be used to paginate through the result set while retaining the `where.before` (timestamp value).

The top level `before` and `after` are added automatically when we specify `@paginate` in GraphQL. The `after` cursor gets the next page and `before` gets the previous. We have the same in `findAuthTokensByUser`.

In practice we change the `where.before` value to page through the results instead of using the cursor. I implemented it this way because we'll need to support that if we ever implement the pinning services API.

Reply:
We could rename this to `createdBefore`?

Reply:
I see, thanks for the details. `createdBefore` 👍🏼